src/services/user.service.js

```diff
@@ -0,0 +1,42 @@
+import { ApiError } from "../exeptions/api.error.js";
+import { User } from "../models/user.js";
+import { emailService } from "../services/email.service.js";
+import { v4 as uuidv4 } from 'uuid';
+
+function getAllActivated() {
+    return User.findAll({
+      where: {
+        activationToken: null
+      }
+    });
+}
+
+function normalize({ id, email }) {
+  return { id, email };
+}
+
+function findByEmail(email) {
+  return User.findOne({where: { email }})
+}
+
+async function register(email, password) {
+  const activationToken = uuidv4(); // token generation
+
+  const existUser = await findByEmail(email);
+
+  if (existUser) {
+    throw  ApiError.badRequest('User already exists', {
```

Comment:
The error message 'User already exists' is repeated in both the main message and the errors object. Consider simplifying the error handling by using a more generic message in the main error and providing specific details in the errors object if necessary.